packages/session-replay-browser/src/config/joined-config.ts

```diff
@@ -22,41 +23,83 @@ export class SessionReplayJoinedConfigGenerator {
     });
   }
 
-  async generateJoinedConfig(sessionId?: number) {
+  async generateJoinedConfig(sessionId?: number): Promise<SessionReplayJoinedConfig> {
     const config: SessionReplayJoinedConfig = { ...this.localConfig };
     // Special case here as optOut is implemented via getter/setter
     config.optOut = this.localConfig.optOut;
+    // We always want captureEnabled to be true, unless there's an override
+    // in the remote config.
+    config.captureEnabled = true;
+    let remoteConfig: SessionReplayRemoteConfig | undefined;
     try {
-      const samplingConfig =
-        this.remoteConfigFetch &&
-        ((await this.remoteConfigFetch.getRemoteConfig(
-          'sessionReplay',
-          'sr_sampling_config',
-          sessionId,
-        )) as SessionReplayRemoteConfig['sr_sampling_config']);
+      if (!this.remoteConfigFetch) {
+        return config;
+      }
 
-      if (samplingConfig && Object.keys(samplingConfig).length > 0) {
-        if (Object.prototype.hasOwnProperty.call(samplingConfig, 'capture_enabled')) {
-          config.captureEnabled = samplingConfig.capture_enabled;
-        } else {
-          config.captureEnabled = false;
-        }
+      const samplingConfig = await this.remoteConfigFetch.getRemoteConfig(
```

Comment:
@kwalker3690 I made some typing changes here that fixed my issue, but this also provides strong typing in general without type assertions. LMK what you think!